source/manual/related-links-on-govuk.md

```diff
@@ -0,0 +1,72 @@
+- Requests need to have the header `Govuk-Use-Recommended-Related-Links` header set to `True` in order to show suggested related links - [this is set by the CDN for all requests](https://github.com/alphagov/govuk-cdn-config/blob/master/vcl_templates/www.vcl.erb#L238).
+
+
+## Concourse pipeline details
+
+Related links are generated and ingested via the following jobs, which are defined in the [govuk-related-links pipeline on Concourse](https://cd.gds-reliability.engineering/teams/govuk-tools/pipelines/govuk-related-links).
+
+### `link-generation`
+
+This job runs the process to generate new related links for a subset of pages on GOV.UK, via the `provision-generation-machine` and `run_link_generation` scripts, which in turn triggers the Python script `run_all.py`.
+
+The related links generated from this process are stored in an S3 environment-specific `govuk-related-links` bucket.
+
+Currently, this job is scheduled to run once every three weeks via a `time` resource attached to the production `run-generation-production` step.
+
+### `link-ingestion`
+
+This job runs the process of ingesting related links via the Publishing API.
+
+Links are retrieved from the environment-specific S3 bucket mentioned previously and then fed (in batches) to the Publishing API. There is a 20 minute delay between batches.
+
+Currently, this job is triggered manually after reviewing a spreadsheet of the generated related links; it will be scheduled to run automatically in the near future.
+
+## Rolling back Links
+
+Should we ever need to roll back any suggested related links, there are a number of ways we can do this.
+
+### Links for individual pages
+
+To rollback suggested related links for an individual page (to show the original related links), use the [Run rake task job on Jenkins](https://deploy.publishing.service.gov.uk/job/run-rake-task/) with the following parameters set:
+  - `TARGET_APPLICATION: publishing-api`
+  - `MACHINE_CLASS: publishing_api`
+  - `RAKE_TASK: content:reset_related_links_for_pages['CONTENT_ID’]`, where `CONTENT_ID` is the content id of the page to remove suggested links from
+
+### Links for certain document types
+
+To exclude a document type or page from having links generated _to_ it, _from_ it, or both, [update the exclusions in the GitHub repository](https://github.com/alphagov/govuk-related-links-recommender/tree/master/src/config). Note: this will take effect from the _next_ time the related links generation process is run, and will not affect current links.
+
+### Suspending all suggested related links
+
+To immediately and temporarily stop showing all suggested related links (for example, if a problem has led to bad links being generated for a large percentage of content / high traffic content), you will need to update the CDN to set the `Govuk-Use-Recommended-Related-Links` header to `False` and then [deploy the CDN](https://deploy.publishing.service.gov.uk/job/Deploy_CDN/) for `vhost: www`.
```

Comment:
Will we also need to invalidate the cache, or is the value of the `Govuk-Use-Recommended-Related-Links` header taken into account in the caching?

Reply:
We set the `Vary` header to use `Govuk-Use-Recommended-Related-Links`, so we will get back different results from origin when we change the header. I was under the belief that if the CDN doesn't have a request with a particular header set when it receives a request, it will forward it on to origin - is there anywhere that we control this more finely?